js/AudioWorkletNode.js

```diff
@@ -0,0 +1,121 @@
+const fs = require('node:fs');
+const path = require('node:path');
+const {
+  Worker,
+} = require('node:worker_threads');
+
+/* eslint-disable no-unused-vars */
+const conversions = require('webidl-conversions');
+const {
+  throwSanitizedError,
+} = require('./lib/errors.js');
+const {
+  kNapiObj,
+} = require('./lib/symbols.js');
+
+/* eslint-enable no-unused-vars */
+
+const AudioNode = require('./AudioNode.js');
+
+module.exports = (jsExport, nativeBinding) => {
+  class AudioWorkletNode extends AudioNode {
+
+    #worker = null;
+    #parameters = {};
+
+    constructor(context, name, options) {
+
+      if (arguments.length < 2) {
+        throw new TypeError(`Failed to construct 'AudioWorkletNode': 2 arguments required, but only ${arguments.length} present`);
+      }
+
+      if (!(context instanceof jsExport.BaseAudioContext)) {
+        throw new TypeError(`Failed to construct 'AudioWorkletNode': argument 1 is not of type BaseAudioContext`);
+      }
+
+      // parsed version of the option to be passed to NAPI
+      const parsedOptions = {};
```

Comment:
@b-ma If you have the time I would really appreciate if you could sanitize the options and use them on the Rust side. We can then remove the default option set from https://github.com/ircam-ismm/node-web-audio-api/pull/124/files#diff-5221df6d2a274b6c20b0dd35dbfbfd3a4a78ff08a5cdfbc032edd0e7ae72ef15R182
The type `AudioWorkletNodeOptions<()>` is still suitable because we move the processorOptions to the Worker on the JS side already.

Reply:
Ok, the thing is done on the JS side, to be continued

(I also have a rather interesting prototype for the `addModule` thing in the pipes :)

Reply:
Done!